### ISSUE TITLE
Hang on Getting NiceHash SMA information Again

### ISSUE THREAD

**Issue by sergentum** [nicehash/NiceHashMiner#551]
tried versions:
NHML-1.8.1.3-Pre2
NHML-1.8.1.4-Pre2
NHML-1.8.1.5
and get all the same
stuck at getting SMA and not starting at all

**Comment by Frageron** [comment]
I started having same problem on 1.8.1.4 today. Tried 1.8.1.5 and 1.8.0.2 they have same issue.
Tried NHM 2.0.1.4, it works.

**Comment by sergentum** [comment]
yeah, i read previous issue with nothing-explaining-title and switched to 2 version, but i didn't like it when tried couple of months ago. But now it's working at least

**Comment by Frageron** [comment]
Yeah, i don't like 2 version too, but i don't know how to get NHML to work again.

**Comment by Frageron** [comment]
NHM 2.0.1.4 don't work too, its just saying that mining is active, but nothing mining.

**Comment by sergentum** [comment]
same problem to me.  i use it on server with 2 xeon, thought that it earns so little that it shows 0 but also i see nothing at the webpage

**Comment by Frageron** [comment]
1.8.1.5 working now

**Comment by DillonN** [comment]
@sergentum is your issue fixed? Looks like it was a temporary server problem

**Comment by sergentum** [comment]
@DillonN i'm sure that is server problem but it appears too frequently, i think nicehash team should do smth with it.

**Comment by sergentum** [comment]
@DillonN about your question - no, i can't start this day again.

**Comment by Doldo** [comment]
Have 2 rigs at home, both of them connected to the same router, same problem, cannot get SMA data and cannot mine on one of those.

**Comment by Sav87** [comment]
Add pull requests #625

**Comment by MSukharev** [comment]
I'v same problem with two rigs. Pings to nicehash.com is o'k, Cast XMR Miner for Vega is working well.

**Comment by dslutej** [comment]
[21:17:34] [SOCKET] 11/28/2017 21:17:34|Debug|WebSocket.sendHttpRequest|A request to the server:
                          GET /v2/nhm HTTP/1.1
                          User-Agent: websocket-sharp/1.0
                          Host: api.nicehash.com
                          Upgrade: websocket
                          Connection: Upgrade
                          Sec-WebSocket-Key: bBXx2RW/+kp6EWLJprcVrQ==
                          Sec-WebSocket-Version: 13
[21:17:34] [SOCKET] 11/28/2017 21:17:34|Debug|WebSocket.sendHttpRequest|A response to this request:
                          HTTP/1.1 503 Service Temporarily Unavailable
                          Date: Tue, 28 Nov 2017 20:17:37 GMT
                          Content-Type: text/html
                          Transfer-Encoding: chunked
                          Connection: keep-alive
                          Set-Cookie: __cfduid=d9dc53ddff95c03a65e5a1d36515169e41511900257; expires=Wed, 28-Nov-18 20:17:37 GMT; path=/; domain=.nicehash.com; HttpOnly
                          Server: cloudflare-nginx
                          CF-RAY: 3c500301ce738649-ARN
[21:17:34] [SOCKET] 11/28/2017 21:17:34|Fatal|WebSocket.connect|Not a WebSocket handshake response.
[21:17:34] [SOCKET] 11/28/2017 21:17:34|Debug|WebSocket.closeHandshake|Was clean?: False
                            sent: False
                            received: False
[21:17:34] [SOCKET] Connection closed code 1002: An error has occurred while connecting.

**Comment by sergentum** [comment]
i can't get why the developers didn't fix this problem long ago

**Comment by skeef** [comment]
Same problem:
`[10:50:49] [SOCKET] 11/30/2017 10:50:49|Debug|WebSocket.sendHttpRequest|A request to the server:
                          GET /v2/nhm HTTP/1.1
                          User-Agent: websocket-sharp/1.0
                          Host: api.nicehash.com
                          Upgrade: websocket
                          Connection: Upgrade
                          Sec-WebSocket-Key: EVjiXsAn7yj1rIgUYvBcTw==
                          Sec-WebSocket-Version: 13
[10:50:49] [SOCKET] 11/30/2017 10:50:49|Debug|WebSocket.sendHttpRequest|A response to this request:
                          HTTP/1.1 503 Service Temporarily Unavailable
                          Date: Thu, 30 Nov 2017 07:50:49 GMT
                          Content-Type: text/html
                          Transfer-Encoding: chunked
                          Connection: keep-alive
                          Set-Cookie: __cfduid=d80c8ef504334093b8979983922c3df461512028249; expires=Fri, 30-Nov-18 07:50:49 GMT; path=/; domain=.nicehash.com; HttpOnly
                          Server: cloudflare-nginx
                          CF-RAY: 3c5c37cd9d3175dc-ARN
[10:50:49] [SOCKET] 11/30/2017 10:50:49|Fatal|WebSocket.connect|Not a WebSocket handshake response.
[10:50:49] [SOCKET] 11/30/2017 10:50:49|Debug|WebSocket.closeHandshake|Was clean?: False
                            sent: False
                            received: False
[10:50:49] [SOCKET] Connection closed code 1002: An error has occurred while connecting.
[10:50:50] [SOCKET] 11/30/2017 10:50:50|Debug|WebSocket.sendHttpRequest|A request to the server:
                          GET /v2/nhm HTTP/1.1
                          User-Agent: websocket-sharp/1.0
                          Host: api.nicehash.com
                          Upgrade: websocket
                          Connection: Upgrade
                          Sec-WebSocket-Key: EVjiXsAn7yj1rIgUYvBcTw==
                          Sec-WebSocket-Version: 13
[10:50:50] [SOCKET] 11/30/2017 10:50:50|Debug|WebSocket.sendHttpRequest|A response to this request:
                          HTTP/1.1 503 Service Temporarily Unavailable
                          Date: Thu, 30 Nov 2017 07:50:50 GMT
                          Content-Type: text/html
                          Transfer-Encoding: chunked
                          Connection: keep-alive
                          Set-Cookie: __cfduid=d2bec855933ff4709cb432b531ddb8cc11512028250; expires=Fri, 30-Nov-18 07:50:50 GMT; path=/; domain=.nicehash.com; HttpOnly
                          Server: cloudflare-nginx
                          CF-RAY: 3c5c37d6c80f86c1-ARN
[10:50:50] [SOCKET] 11/30/2017 10:50:50|Fatal|WebSocket.connect|Not a WebSocket handshake response.
[10:50:50] [SOCKET] 11/30/2017 10:50:50|Debug|WebSocket.closeHandshake|Was clean?: False
                            sent: False
                            received: False
[10:50:50] [SOCKET] Connection closed code 1002: An error has occurred while connecting.`

**Comment by sergentum** [comment]
guys, i localized problem a bit. any windows proxy or vpn can help. So i guess there is some troubles with my provider or government blocks some ip adresses

**Comment by Sav87** [comment]
added an alternate way to get sma

**Comment by xtrime-ru** [comment]
@Sav87 made few fixes, so NHML can use another way of getting SMA data if primary method fails.
I've compiled his fixes and made this release:

https://github.com/xtrime-ru/NiceHashMinerLegacy-SMA-FIX/releases

**Comment by nerdatwork** [comment]
Hate the SMA issue

**Comment by ekadanthcomputers** [comment]
I have 5 rigs and I was faced with this issue too. I deleted nicehash, cleared cache and had to benchmark one of my rigs all again but this did not resolve the issue. I tried the step with disabling IPv6 to no avail. I turned off all my rigs. Switched on one and changed the 'Service Location' and 'Currency' (in settings) on  the NiceHash Miner. This resolved the issue for me. All my rigs are working again.
Please let me know if this works for you.

**Comment by mascondante** [comment]
Same here. Didn't have a problem on first load but on the second load a couple days later I'm in the same situation. If it is a censorship issue I am in the eastern United States if that helps narrow things down. I never did change my server location and now even after deleting the install and making a new one it still hangs at the same point.

**Comment by xtrime-ru** [comment]
@mascondante NiceHash now on heavy maintaince. Even site is down. Api, aps, pools doesn't work for last 3 hours. This is not software issue.

**Comment by ghost** [comment]
@xtrime-ru dont lie to me kid

**Comment by xtrime-ru** [comment]
@dan28453 Wat?! :)

**Comment by ghost** [comment]
@xtrime-ru I dunno it was my cat

**Comment by darkneon2002** [comment]
2.0.1.6 beta, has SMA issues once again :(

**Comment by Joe-J-Garcia** [comment]
2.0.16 and 2.018 SMA issue from here

**Comment by swift2geek** [comment]
1.8.1.6 same problem again!!!! please do something!

**Comment by xtrime-ru** [comment]
@Sav87 Can you merge your fixes with 1.8.1.6 release and compile NHML without dependencies?

**Comment by pjdouillard** [comment]
But what is this query that the client needs?

**Comment by romin991** [comment]
i got this issue now.. so it's all server fault?

**Comment by xtrime-ru** [comment]
@romin991 No, probably its a  3d party ddos protection issue (cloudflare). I downgraded all my rigs to fixed 1.8.1.5, you can get it in my repo.
Fixed exe get profability data via old request if new method fails.

**Comment by AtlAntA118** [comment]
Man is this problem getting annoying. Been trying to get my rig started for over an hour now...

**Comment by BCtopics** [comment]
Having this problem as well.

**Comment by cmbconcretefire** [comment]
@DillonN , Hey fellas. Add me to the list. However, I have a possible clue. I fixed mine. I am lucky enough to have 2 rigs  so I can compare. The Devs won't believe me, YOU won't believe me....I don't care. I live in the US. Therefor I have "San Jose" as my location to connect to the server on BOTH rigs. Yes, I am 100% positive. 

So today, it just so happened that I was messing around with MSI afterburner and adjusting some settings WHILE NHML was mining. I also have the "run OnGPULost.bat" setting ticked ON in the settings section. Well......sure enough, my messing around caused one of my GPU's to drop out. And I thought "Good, let's see if this whole 'fix lost gpu thing works." ...... I watched the whole process. It shut down NHML, the screen flickered < I assume because it was reviving a lost GPU, > the screen came back to life and sure enough, NHML opened ....... BUT GOT STUCK on the SMA thing. 

I even restarted my rig. Then I noticed THE PROBLEM...... somehow it had switched my server location from San Jose to Amsterdam. ?????? I looked at my other rig, inside the configs folder and instinctively looked at the "General.json" file. Inside there, I found this >  "ServiceLocation": 1, 

So I looked back on the rig that was stuck and that General.json file had "ServiceLocation": 0,

I used task manager to force NHML closed, changed "ServiceLocation": 0, to "ServiceLocation": 1, 
and double clicked on the NHML.exe. Connected straight away. 

Don't care if you don't believe me. Or say that can't happen. It did happen. And that's how I fixed it. So a simple fix for this would be a backup .bat file / script that runs a BACKUP "General.json" on startup to ensure that the correct settings / even if somehow corrupted or changed / will be automatically re-populated into the MAIN "general.json" file. My 2 cents. Peace.
